docs/src/prototypes/customToolbar/CustomToolbar.tsx

```diff
@@ -0,0 +1,230 @@
+
+    {
+      key: 'more',
+      icon: {
+        name: 'more',
+        size: 'large',
+      },
+      onClick: () => _.invoke(props, 'onMoreClick'),
+      variables: { isCtItemPrimary: true },
+    },
+  ].filter(Boolean)
+
+const sidebarButtons: CustomToolbarLayout = props => [
+  {
+    active: props.sidebarSelected === 'chat',
+    icon: {
+      name: 'chat',
+      outline: 'true',
+      size: 'large',
+    },
+    key: 'chat',
+    onClick: () =>
+      _.invoke(props, 'onSidebarChange', props.sidebarSelected === 'chat' ? false : 'chat'),
+    variables: { isCtItemWithNotification: props.chatHasDot, isCtItemIconNoFill: true },
+  },
+  {
+    active: props.sidebarSelected === 'participant-add',
+    icon: {
+      name: 'participant-add',
+      outline: 'true',
+      size: 'large',
+    },
+    key: 'participant-add',
+    onClick: () =>
+      _.invoke(
+        props,
+        'onSidebarChange',
+        props.sidebarSelected === 'participant-add' ? false : 'participant-add',
+      ),
+    variables: { isCtItemIconNoFill: true },
+  },
+]
+
+const layoutItems = {
+  endCall: props => ({
+    key: 'end-call',
+    icon: {
+      name: 'call-end',
+      size: 'large',
+    },
+    onClick: () => _.invoke(props, 'onEndCallClick'),
+    variables: { isCtItemDanger: true },
+  }),
+}
+
+const layouts: Record<CustomToolbarProps['layout'], CustomToolbarLayout> = {
+  standard: props => [...commonLayout(props), ...sidebarButtons(props), layoutItems.endCall(props)],
+
+  'desktop-share': props => [
+    ...commonLayout(props),
+    ...sidebarButtons(props),
+    { key: 'divider-sidebar', kind: 'divider' },
+    {
+      key: 'stop-sharing',
+      kind: 'custom',
+      content: <Button content="Stop Sharing" />,
+    },
+
+    layoutItems.endCall(props),
+  ],
+  'powerpoint-presenter': props => [
+    ...commonLayout(props),
+    ...sidebarButtons(props),
+    { key: 'divider-sidebar', kind: 'divider' },
+
+    {
+      key: 'stop-sharing',
+      icon: {
+        name: 'call-control-stop-presenting-new',
+        size: 'large',
+      },
+      onClick: () => _.invoke(props, 'onStopSharingClick'),
+    },
+
+    {
+      key: 'ppt-prev',
+      icon: {
+        name: 'chevron-down',
+        rotate: 90,
+        outline: true,
+      },
+      onClick: () => _.invoke(props, 'onPptPrevClick'),
+    },
+
+    {
+      key: 'ppt-slide-number',
+      kind: 'custom',
+      fitted: true,
+      content: <Text size="small">{props.pptSlide}</Text>,
+    },
+
+    {
+      key: 'ppt-next',
+      icon: {
+        name: 'chevron-down',
+        rotate: -90,
+        outline: true,
+      },
+
+      onClick: () => _.invoke(props, 'onPptNextClick'),
+    },
+
+    layoutItems.endCall(props),
+  ],
+}
+
+const CustomToolbar: React.FunctionComponent<CustomToolbarProps> = props => {
+  const { layout = 'standard' } = props
+
+  return (
+    <Toolbar
+      variables={{ dividerMargin: 0, borderRadius: 0, itemHeight: '4rem', isCt: true }}
```

Comment:
I don't like that we have these defined in the tsx file. Can we move them to variables?

Reply:
Variables removed.
That means built-in variables are useless for (conditional) style overrides :-(